### PR TITLE
disable spring cloud discovery when running test to speed up test pro…

### LINKED ISSUE #4603: test: speed up SpringBootTest by close spring cloud discovery
- [x] I have checked the [discussions](https://github.com/ctripcorp/apollo/discussions)
- [x] I have searched the [issues](https://github.com/ctripcorp/apollo/issues) of this repository and believe that this is not a duplicate.
- [x] I have checked the [FAQ](https://www.apolloconfig.com/#/zh/faq/common-issues-in-deployment-and-development-phase) of this repository and believe that this is not a duplicate.

**Describe the bug**
Some tests run a little slow.

**To Reproduce**
1. run com.ctrip.framework.apollo.biz.service.AccessKeyServiceTest

log

```java
[WARN] org.springframework.cloud.loadbalancer.config.LoadBalancerCacheAutoConfiguration$LoadBalancerCaffeineWarnLogger - Spring Cloud LoadBalancer is currently working with the default cache. While this cache implementation is useful for development and tests, it's recommended to use Caffeine cache in production.You can switch to using Caffeine cache, by adding it and org.springframework.cache.caffeine.CaffeineCacheManager to the classpath.
[WARN] com.netflix.discovery.shared.transport.decorator.RetryableEurekaHttpClient - Request execution failed with message: I/O error on GET request for "http://localhost:8761/eureka/apps/": Connect to localhost:8761 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused: connect; nested exception is org.apache.http.conn.HttpHostConnectException: Connect to localhost:8761 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused: connect
[WARN] com.netflix.discovery.DiscoveryClient - Using default backup registry implementation which does not do anything.
[WARN] com.netflix.discovery.shared.transport.decorator.RetryableEurekaHttpClient - Request execution failed with message: I/O error on POST request for "http://localhost:8761/eureka/apps/UNKNOWN": Connect to localhost:8761 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused: connect; nested exception is org.apache.http.conn.HttpHostConnectException: Connect to localhost:8761 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused: connect
[WARN] com.netflix.discovery.DiscoveryClient - DiscoveryClient_UNKNOWN/LAPTOP-PJ6N2E8S:0 - registration failed Cannot execute request on any known server
com.netflix.discovery.shared.transport.TransportException: Cannot execute request on any known server
	at com.netflix.discovery.shared.transport.decorator.RetryableEurekaHttpClient.execute(RetryableEurekaHttpClient.java:112)
	at com.netflix.discovery.shared.transport.decorator.EurekaHttpClientDecorator.register(EurekaHttpClientDecorator.java:56)
	at com.netflix.discovery.shared.transport.decorator.EurekaHttpClientDecorator$1.execute(EurekaHttpClientDecorator.java:59)
	at com.netflix.discovery.shared.transport.decorator.SessionedEurekaHttpClient.execute(SessionedEurekaHttpClient.java:77)
	at com.netflix.discovery.shared.transport.decorator.EurekaHttpClientDecorator.register(EurekaHttpClientDecorator.java:56)
	at com.netflix.discovery.DiscoveryClient.register(DiscoveryClient.java:876)
	at com.netflix.discovery.InstanceInfoReplicator.run(InstanceInfoReplicator.java:121)
	at com.netflix.discovery.InstanceInfoReplicator$1.run(InstanceInfoReplicator.java:101)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
[WARN] com.netflix.discovery.InstanceInfoReplicator - There was a problem with the instance info replicator
```

This won't cause test failure, but we can speed up the test by close it.

**Expected behavior**
Test should run as fast as possible.

**Screenshots**
None.

### Additional Details & Logs

- Version 2.1.0-SNAPSHOT
- Error logs
- Configuration
- Platform and Operating System: Win11 21H1

### one of solution

Some tests doesn't need to use registy, we can disable it by add

```
spring.cloud.discovery.enabled=false
```

in `@SpringBootTest` properties or other place.

On my own machine, it can decrease test time consuming from 22s to 12s.

### PR DESCRIPTION
**Pull request by wutingjia**
## What's the purpose of this PR
disable spring cloud discovery when running test to speed up test process and reduce error log

## Which issue(s) this PR fixes:
Fixes #4603

## Brief changelog

disable spring cloud discovery when running test to speed up test process and reduce error log

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Read the [Contributing Guide](https://github.com/ctripcorp/apollo/blob/master/CONTRIBUTING.md) before making this pull request.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit tests to verify the code.
- [x] Run `mvn clean test` to make sure this pull request doesn't break anything.
- [x] Update the [`CHANGES` log](https://github.com/ctripcorp/apollo/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by wutingjia** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by wutingjia** [comment]
recheck

File: CHANGES.md [hunk 1]
```diff
@@ -29,9 +29,10 @@ Apollo 2.1.0
 * [fix openapi item with url illegalKey 400 error](https://github.com/apolloconfig/apollo/pull/4549)
 * [fix the exception occurred when publish/rollback namespaces with grayrelease](https://github.com/apolloconfig/apollo/pull/4564)
 * [fix create namespace with single dot 500 error](https://github.com/apolloconfig/apollo/pull/4568)
+* [Add overloaded shortcut method to register BeanDefinition](https://github.com/apolloconfig/apollo/pull/4574)
 * [Add nodejs client sdk and fix doc](https://github.com/apolloconfig/apollo/pull/4590)
 * [Move apollo-core, apollo-client, apollo-mockserver, apollo-openapi and apollo-client-config-data to apollo-java repo](https://github.com/apolloconfig/apollo/pull/4594)
 * [fix get the openapi interface that contains namespace information for deleted items](https://github.com/apolloconfig/apollo/pull/4596)
-
+* [disable spring cloud discovery when running test to speed up test process and reduce error log](https://github.com/apolloconfig/apollo/pull/4604)
```
Comment (Anilople) [inline CHANGES.md:36]: test's changelog doesn't need to add to CHANGES.md

**Review by Anilople — CHANGES_REQUESTED** [review]

File: CHANGES.md (hunk 1 above)
Comment (wutingjia) [reply CHANGES.md:36]: ok

**Review by Anilople — APPROVED** [review]
LGTM

**Pull request merged**